packages/blocks/src/api/registration.js

```diff
@@ -310,6 +316,114 @@ export function registerBlockType( name, settings ) {
 	return settings;
 }
 
+/**
+ * Translates block settings provided with metadata using the i18n schema.
+ *
+ * @param {string|string[]|Object[]} i18nSchema    I18n schema for the block setting.
+ * @param {string|string[]|Object[]} settingValue  Value for the block setting.
+ * @param {string}                   textdomain    Textdomain to use with translations.
+ *
+ * @return {string|string[]|Object[]} Translated setting.
+ */
+function translateBlockSettingUsingI18nSchema(
+	i18nSchema,
+	settingValue,
+	textdomain
+) {
+	if ( isString( i18nSchema ) && isString( settingValue ) ) {
+		// eslint-disable-next-line @wordpress/i18n-no-variables, @wordpress/i18n-text-domain
```

Comment:
@swissspidy, do we need to do something special to ensure this particular translation call is ignored during WP-CLI processing?

Reply:
String extraction will skip anything that‘s not a string literal, so no.

Reply:
Thanks!

Reply:
@gziolo How are the translations supposed to be in `wp.i18n` without a specific `wp.i18n.setLocaleData()` call? The strings from the block.json aren't assigned to a JavaScript file thus not automatically part of a JSON translation file. But I might be missing something here? Or are we fully relying on server-side registration for the strings?

Reply:
In the context of WordPress core, they are always registered on the server and translated there before passed to the client. At the moment, there is a temporary JS method `unstable__bootstrapServerSideBlockDefinitions` that takes care of setting data generated with PHP. In the future, we are going to use the REST API endpoint for blocks, but the general idea is going to be the same - translations get applied when registering a block on the server.

In practice, this implementation is a fallback for everything else that depends only on JavaScript:
- standalone block editor like Storybook integration (although this is a bad example as it is never translated)
- native mobile apps still consume only JavaScript codebase so they have to collect all translations themselves